---
title: "01-deseq2"
output: html_document
---

```{r setup, include=FALSE}
# https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
library(tidyverse)
library(DESeq2)
library(cowplot)

source("functions.R")

knitr::opts_chunk$set(fig.path = '../figures/',
                      message=FALSE, warning=FALSE)
```


Now, I'll do RNAseq on all the data

```{r deseq2}
dds <- DESeqDataSetFromMatrix(countGonad, colGonad, ~ sex*treatment)
dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
vsd <- vst(dds, blind=FALSE) ## variance stabilized
head(assay(vsd),3)
#write.csv(assay(vsd), file = "../results/01-vsd.csv", row.names = T)
```

```{r res}
res_summary <- function(mycontrast){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(summary(res))
  #print(head((res[order(res$padj),]), 3))
  cat("\n")
}

res_summary(c("sex", "female", "male"))
res_summary(c("treatment", "hatch", "control"))
```

```{r pca}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("sex","treatment"), returnData=TRUE)

percentVar <- round(100 * attr(pcadata, "percentVar"))
pcadata$sextissue <- as.factor(paste(pcadata$sex, pcadata$tissue, sep="_"))
pcadata$sextreatment <- as.factor(paste(pcadata$sex, pcadata$treatment, sep="_"))


summary(aov(PC1 ~ sex * treatment * tissue, data=pcadata)) 
summary(aov(PC2 ~ sex * treatment * tissue, data=pcadata)) 

PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, color=sextreatment, shape= tissue)) +
    geom_point(size=1.5) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    stat_ellipse(level = 0.95) + 
    theme_cowplot(font_size = 7, line_size = 0.25)  
PCA12

pdf(file="../figures/01-pcaplot12.pdf", width=3.3, height=2.5)
plot(PCA12)
dev.off()

```





Now, I'll do RNAseq on all the data

```{r deseq2}
dds <- DESeqDataSetFromMatrix(countHyp, colHyp, ~ treatment)
dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
vsd <- vst(dds, blind=FALSE) ## variance stabilized
#head(assay(vsd),3)
#write.csv(assay(vsd), file = "../results/01-vsd.csv", row.names = T)
```

```{r res}
res_summary <- function(mycontrast){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(summary(res))
  #print(head((res[order(res$padj),]), 3))
  cat("\n")
}

res_summary(c("sex", "female", "male"))
res_summary(c("treatment", "hatch", "control"))
```

```{r pca}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("treatment"), returnData=TRUE)

percentVar <- round(100 * attr(pcadata, "percentVar"))

summary(aov(PC1 ~ treatment, data=pcadata)) 
summary(aov(PC2 ~  treatment , data=pcadata)) 

PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, color=treatment)) +
    geom_point(size=1.5) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    stat_ellipse(level = 0.95) + 
    theme_cowplot(font_size = 7, line_size = 0.25)  +
  labs(title ="Female Hypothalamus")
PCA12


```
